src/qudi/core/modulemanager.py

```diff
@@ -246,6 +248,74 @@ def _qudi_main_ref_dead_callback(self):
                      'ModuleManager.')
         self.clear()
 
+    def dump_status_variables(self):
+        """
+        Method that dumps the status variables of all active modules.
+        """
+        logger.debug(f"Dumping status variables")
+        with self._lock:
+            for _, module in self.modules.items():
+                if module.is_active:
+                    module.instance._dump_status_variables()
+
+    def toggle_automated_status_variable_dumping(self, toggle):
+        """
+        Method that creates or destroys the QTimer for handling the automatic dumping of status variables.
+
+        @param bool toggle: boolean that determines whether timer is created or destroyed.
+        """
+        if not toggle:
+            logger.info(f"Automated status variable saving disabled.")
+            self.automated_status_variable_dumping_timer.stop()
+            try:
+                self.automated_status_variable_dumping_timer.timeout.disconnect()
+            except RuntimeError:
```

Comment:
What's the origin of the RuntimeError?

Reply:
If the timer was already destroyed (or never activated), calling the disconnect function will result in a RuntimError. To not have this throw an Error during e.g. shutdown of Qudi, just do nothing.